### ISSUE TITLE
WTT hangs

### ISSUE THREAD

**Issue by registriren** [Neolysion/WTT-Bridge#10]
Some (unsupported, perhaps from groups with a higher number of participants) messages WA, especially if a lot of them, you call the stop service WTT because of an error. Once you restart the WTT to success does not. I manually start the yowsup several times, get the error, repeat it 10 times and all starts to work. The problem messages disappear, but the following simple messages are delivered correctly. Next, the WTT is working again. Can I automate this process so that in the event of such error WTT he restarted the problematic reception of messages until the error does not stop?

the format of the error in the log file:
[log3.txt](https://github.com/Neolysion/WTT-Bridge/files/3244363/log3.txt)

**Comment by s-boris** [comment]
this seems to be an error in yowsup. I don't think I can catch it in WTT.  
You could try to delete the axolot.db in your yowsup profile (~/.config/yowsup/[phonenumber]). This will cause yowsup to regenerate the keys on next startup, maybe it will fix the problem.

Otherwise I don't think there is much we can do but wait until the new yowsup version is out.

**Comment by registriren** [comment]
tonight WTT froze up again the log file stops:
[log-stop-WTT.txt](https://github.com/Neolysion/WTT-Bridge/files/3245357/log-stop-WTT.txt)

restarting WTT solved the problem:
[log-restart.txt](https://github.com/Neolysion/WTT-Bridge/files/3245358/log-restart.txt)

how to determine that WTT is frozen? how to automate its restart?

**Comment by registriren** [comment]
> this seems to be an error in yowsup. I don't think I can catch it in WTT.
> You could try to delete the axolot.db in your yowsup profile (~/.config/yowsup/[phonenumber]). This will cause yowsup to regenerate the keys on next startup, maybe it will fix the problem.
> 
> Otherwise I don't think there is much we can do but wait until the new yowsup version is out.

axolot.db updated, I will observe what to change

**Comment by registriren** [comment]
WTT-service again hung, and he stopped regardless of the error yowsup.
a log of the last run:
[log5.txt](https://github.com/Neolysion/WTT-Bridge/files/3250650/log5.txt)
service worked less than 10 hours ((

**Comment by LevinGermann** [comment]
Hmm, I would propose to wait for a new yowsup-release, the yowsup dev mentioned a couple of times that this issue has been fixed and will be included in the next release.
If you want to try on the current master branch of yowsup, @Neolysion has included support for it in the WTT dev branch.
You can check it out ("git checkout dev" in the WTT-Bridge directory)
git clone git@github.com:tgalal/yowsup.git
run "python setup.py install" and remove the current yowsup installation, so that WTT is running with the latest "unreleased" yowsup version...

**Comment by registriren** [comment]
I'm not sure wt always stops because of yowsup ([log5.txt](https://github.com/Neolysion/WTT-Bridge/files/3252168/log5.txt)), but it is not so important. A universal solution would be to automatically restart the WTT if it stopped. This will allow you to calmly wait for the new release of yowsup.

**Comment by s-boris** [comment]
@registriren are you sure that WTT or yowsup stops completely? because after those errors it is still printing the state transition messages which come from yowsup.
I just added an autorestart script that will run WTT again if it crashes.

Anyway, I would suggest that you schedule the service to autorestart (cronjob) every few hours, if you really want to use it  in its current state.

**Comment by registriren** [comment]
I don't know the exact reason for the stop WTT. Stop going through about 10 hours of work. The process remains in memory, the log ends with the line:
- transitions.core - INFO - Entered state init

**Comment by registriren** [comment]
> Anyway, I would suggest that you schedule the service to autorestart (cronjob) every few hours, if you really want to use it in its current state.

I agree, I am now helping to write a restart script that analyzes the stop recording log file on the line: "...transitions.core - INFO - Entered state init"

**Comment by registriren** [comment]
WTT still works no more than 10 hours, the reason for the stop is unknown. We have written a script for cron, which controls the stop of the service and restarts it.


[script.zip](https://github.com/Neolysion/WTT-Bridge/files/3261680/script.zip)

**Comment by LevinGermann** [comment]
Nice, I think your scripts work fine and are good as a work-around for now. Thanks for them!

**Comment by registriren** [comment]
Yes, the script works, WTT is automatically restarted. How to find out the reason for the stop?

**Comment by LevinGermann** [comment]
I'm not really sure, I can't find any errors in the log... Maybe WTT needs to have some extended debug logging messages to find out the root cause.

**Comment by registriren** [comment]
in DEBUG mode, the log continues to be written, but the messaging stops at some point, I was unable to identify it.

**Comment by mrsnapznet** [comment]
registriren hi can i contact you for private

**Comment by registriren** [comment]
Yes, please give your contact
